homeassistant/helpers/reload.py

```diff
@@ -120,7 +120,7 @@ async def _async_setup_platform(
         )
         return
 
-    entity_component: EntityComponent = hass.data[integration_platform]
+    entity_component: EntityComponent = hass.data[integration_platform]  # type: ignore[type-arg]
```

Comment:
I've used the base Entity class here because I wasn't quite sure.
Maybe I should be using # type: ignore[var-annotated] instead?